components/Button/Button.test.tsx

```diff
@@ -1,15 +1,32 @@
 import React from 'react';
```

Comment:
This is incorrectly testing the Button. Checking for the Variants is good, but we also need to test when the sizes change and a custom className is added as well. Also, we should not be forcing the buttonVariants() call. It's logic that the Button component will run through. Allow the code to run through the logic instead of manually forcing it. When unit testing, you want the code to run its course end to end. From a testing standpoint, you want to stub the responses so they match what you expect. 

Here's the start I had previously. We'd still need to add another test for the case when a custom className is used.
```
const variants: Array<"link" | "default" | "outline" | "destructive" | "secondary" | "ghost" | null | undefined> = ['default', 'outline', 'destructive', 'secondary', 'ghost', 'link'];
const sizes: Array<"default" | "sm" | "lg" | "icon" | null | undefined> = ['default', 'sm', 'lg', 'icon'];

const variantClasses = {
  default: 'bg-orange-600 text-white hover:bg-orange-600/90',
  outline: 'border border-input bg-background hover:bg-accent hover:text-accent-foreground',
  destructive: 'bg-destructive text-destructive-foreground hover:bg-destructive/90',
  secondary: 'bg-secondary text-secondary-foreground hover:bg-secondary/80',
  ghost: 'hover:bg-accent hover:text-accent-foreground',
  link: 'text-primary underline-offset-4 hover:underline',
};

const sizeClasses = {
  default: 'h-10 px-4 py-2',
  sm: 'h-9 rounded-md px-3',
  lg: 'h-11 rounded-md px-8',
  icon: 'h-10 w-10',
};
describe('Button', () => {
  variants.forEach(variant => {
    sizes.forEach(size => {
      it(`renders correctly with variant ${variant} and size ${size}`, () => {
        render(<Button variant={variant} size={size}>Test Button</Button>);

        const button = screen.getByRole('button');
        expect(button).toBeInTheDocument();
        
        // Check if the button has the correct class for the variant and size
        if (variant) {
          expect(button).toHaveClass(variantClasses[variant]);
        }
        if (size) {
          expect(button).toHaveClass(sizeClasses[size]);
        }
      });
    });
  });
});
```